test/GoogleAnalyticsEventProcessor/buildRequestUrl_test.js

```diff
@@ -0,0 +1,87 @@
+goog.module('measurementLibrary.eventProcessor.testing.GoogleAnalyticsEventProcessor.buildRequestUrl');
+goog.setTestOnly();
+
+const GoogleAnalyticsEventProcessor = goog.require('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+describe('The `buildRequestUrl` method ' +
+    'of GoogleAnalyticsEventProcessor', () => {
+  /**
+   * Tests the `buildRequestUrl` function by passing in an API secret,
+   * measurement ID, and measurement URL to construct the event processor
+   * with and comparing the result of the function with the expected result.
+   * @param {string} apiSecret
+   * @param {string} measurementId
+   * @param {string} measurementUrl
+   * @param {string} expected
+   */
+  function checkRequestUrl(apiSecret, measurementId, measurementUrl, expected) {
+    const eventProcessor = new GoogleAnalyticsEventProcessor({
+      'api_secret': apiSecret,
+      'measurement_id': measurementId,
+      'measurement_url': measurementUrl,
+    });
+
+    expect(eventProcessor.buildRequestUrl())
+        .toBe(expected);
+  }
+
+  it('creates the correct api hit with default url', () => {
+    checkRequestUrl(
+      /** apiSecret */ '123',
+      /** measurementId */ '456',
+      /** measurementUrl */ undefined,
+      /** expected */ 'https://www.google-analytics.com/mp/collect?measurement_id=456&api_secret=123'
```

Comment:
Nit: Line too long.

I usually break URL strings after the "?" first, then after an "&" or "/" if more breaks are required.